src/java.base/share/classes/java/net/URLConnection.java

```diff
@@ -588,6 +588,12 @@ public String getHeaderField(String name) {
      * unmodifiable List of Strings that represents
      * the corresponding field values.
      *
+     * This method is overridden by the subclasses of {@code URLConnection}.
+     *
+     * @implSpec In the implementation of these methods, if a given key has multiple
+     * corresponding values, they must be returned in the order they were added,
+     * preserving the insertion-order.
+     *
```

Comment:
Here the `@implSpec` is right because it describes how subclasses should implement the method.
Maybe there should be an additional `@implNote` as well to describe what the default implementation does.

```
      * @implNote The default implementation of this method returns an empty map always.
```

The `@implNote` is useful for subclasses to decide whether they should override the default implementation.

Reply:
Added, thanks Daniel

Reply:
I believe that "The default implementation ... " here should be part of the implSpec. In many other cases in the Java SE specification implSpec is used to describe the default implementation of a non-abstract member of an interface or abstract class. The implSpec is normative and as such "requires" that the actual implementation behave as such. The implNote is not normative.